src/native/corehost/bundle/file_entry.cpp

```diff
@@ -10,15 +10,26 @@ using namespace bundle;
 
 bool file_entry_t::is_valid() const
 {
-    return m_offset > 0 && m_size >= 0 &&
+    return m_offset > 0 && m_size >= 0 && m_compressedSize >= 0 &&
         static_cast<file_type_t>(m_type) < file_type_t::__last;
 }
 
-file_entry_t file_entry_t::read(reader_t &reader, bool force_extraction)
+file_entry_t file_entry_t::read(reader_t &reader, uint32_t major_version, bool force_extraction)
```

Comment:
```suggestion
file_entry_t file_entry_t::read(reader_t &reader, uint32_t bundle_major_version, bool force_extraction)
```

It's not clear what the version is about (the file itself, or the parent bundle).

Reply:
12690 **[_135600_]()**